test/functional/test_framework/test_framework.py

```diff
@@ -557,20 +557,11 @@ def restart_node(self, i, extra_args=None):
     def wait_for_node_exit(self, i, timeout):
         self.nodes[i].process.wait(timeout)
 
-    def get_node_addr_libp2p(self, a):
-        # Call p2p_get_bind_address just to get node id
-        id_a = self.nodes[a].p2p_get_bind_address().split("/p2p/")[1]
-        return p2p_url(a) + "/p2p/" + id_a
-
-    def get_node_addr_mock(a):
-        return p2p_url(a)
-
     def connect_nodes(self, a, b):
         count_a = self.nodes[a].p2p_get_peer_count()
         count_b = self.nodes[b].p2p_get_peer_count()
 
-        # TODO: Replace with get_node_addr_mock once libp2p removed
-        addr_a = self.get_node_addr_libp2p(a)
+        addr_a = p2p_url(a)
         ret = self.nodes[b].p2p_connect(addr_a)
 
         wait_until_helper(lambda:
```

Comment:
I want to say that while this works now, this can easily lead to a dead-lock. You can image a situation where a connection is lost from one peer and they never reach the +1 value. It's rare, but possible. There should be a better way to do this... though not in this PR. Please create an issue for this.

The better solution would be checking that we're connecting to a peer by some kind of id, as node label. I think bitcoin does something like that. When a node is created and run, it's given a label that's attached to its version that can be used to identify the node.

Reply:
https://github.com/mintlayer/mintlayer-core/issues/624